metals/src/main/scala/scala/meta/internal/process/SystemProcess.scala

```diff
@@ -1,30 +1,80 @@
 package scala.meta.internal.process
 
-import java.nio.file.Path
+import java.io.InputStream
+import java.util.concurrent.TimeUnit
 
-import scala.sys.process._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.jdk.CollectionConverters._
+import scala.sys.process.BasicIO
+import scala.util.control.NonFatal
 
-import scala.meta.internal.metals.Time
-import scala.meta.internal.metals.Timer
-import scala.meta.pc.CancelToken
+import scala.meta.internal.ansi.LineListener
+import scala.meta.io.AbsolutePath
+
+trait SystemProcess {
+  def complete: Future[Int]
+  def cancel: Unit
+}
 
 object SystemProcess {
+
   def run(
-      shortName: String,
-      args: List[String],
-      reproduceArgs: List[String],
-      cwd: Path,
-      token: CancelToken
-  ): Unit = {
-    val exportTimer = new Timer(Time.system)
-    scribe.info(args.mkString("process: ", " ", ""))
-    val exit = Process(args, cwd = Some(cwd.toFile())).!
-    if (exit != 0) {
-      val message = s"$shortName command failed with exit code $exit, " +
-        s"to reproduce run the command below:\n\t${reproduceArgs.mkString(" ")}"
-      throw new MessageOnlyException(message)
-    } else {
-      scribe.info(s"time: ran '$shortName' in $exportTimer")
+      cmd: List[String],
+      cwd: AbsolutePath,
+      redirectErrorOutput: Boolean,
+      env: Map[String, String]
+  ): SystemProcess = {
+    def readOutput(stream: InputStream, f: String => Unit): Thread = {
+      val lineListeneter = new LineListener(f)
+      val thread = new Thread {
+        override def run(): Unit = {
+          // use scala.sys.process implementation
+          try BasicIO.processFully(line => lineListeneter.appendLine(line))(
+            stream
+          )
+          catch { case NonFatal(_) => }
```

Comment:
If I'm understanding this correctly this would probably only ever be an `IOException` right? If that happens do we want to log this out or anything? If this ever fails it might be hard to track down if we do nothing here.

Reply:
Thx for pointing at this! I checked it again. 
I added this `try``catch` by mistake. `IOException` handling is done inside `BasicIO`.

The actual problem was that I caught an exception while I was debugging test on windows but now I see that it came from `ps.start()`.Reworked this thing.

Reply:
All right, it might throw exception. Slightly changed this part.
The funny thing that this comment directed me to cover an additional case)